backend/app/core/providers.py

```
@@ -1,7 +1,10 @@
+import logging
 from typing import Dict, List, Optional
```

Ruff (0.12.2)

`typing.Dict` is deprecated, use `dict` instead

(UP035)